### DISCUSSION TITLE
Conflict Resolution ?

### DISCUSSION THREAD

**Discussion by MentalGear** [payloadcms/payload#1432] [Q&A]
After browsing through the docs, I couldn't find a note on what conflict resolution is used?

Is it last updated .. ? Is there any lock or note when a user tries to access an item that is being edited in the admin UI by another user ?

**Comment by jacobsfletch** [comment]
> Is there any lock or note when a user tries to access an item that is being edited in the admin UI by another user ?

We don't currently have support for this but it's a great idea. WordPress does this nicely, I think we can take queues from them.

Besides that, updates are processed in the order that they arrive.

**Reply by dawidmachon** [reply]
Any changes in this area? Any local api for FindAndUpdate from mongo?

**Reply by DanRibbens** [reply]
First part, not yet. Expect to have some work in this area taking place soon.
Second part, you can use the local update with a `where` query to effectively `findAndUpdate`.

**Reply by dawidmachon** [reply]
> First part, not yet. Expect to have some work in this area taking place soon. Second part, you can use the local update with a `where` query to effectively `findAndUpdate`.

Is it guarantee locking down document?  If two concurrent local updates with where statement happen, in exactly the same moment, but updates will be different. How it will bahavie? I'm worried about the situation when "find" take in the same moment, and update part after it. And one change will be lost. Payload local update with where forces locking down document for that time or not?

**Reply by DanRibbens** [reply]
As jacobsfletch said, the last call to update the DB wins.

**Reply by dawidmachon** [reply]
> As jacobsfletch said, the last call to update the DB wins.

Okay, so it's not like findAndUpdate from mongo.

From mongo blog:

> A [findAndModify](https://www.mongodb.com/docs/manual/reference/command/findAndModify/#mongodb-dbcommand-dbcmd.findAndModify) operation on a document is [atomic](https://www.mongodb.com/docs/manual/reference/glossary/#std-term-atomic-operation): if the find condition matches a document, the update is performed on that document. Concurrent queries and additional updates on that document are not affected until the current update is complete.

**Reply by DanRibbens** [reply]
Yeah, I was thinking your followup question was  unrelated to the conflict question DB operations! Sorry for that confusion.

**Reply by dawidmachon** [reply]
> Yeah, I was thinking your followup question was unrelated to the conflict question DB operations! Sorry for that confusion.

Maybe I was not clear. But now everything is okay ;) Payload CMS is game changer for me as a CMS, but I'm using it for commerce app so atomic operations, transactions, rollbacks are crucial on longer run ;) So probably I picked a wrong tool for that for now, but Payload is so fun to use I will not pass it ; ) There is a lot of room left for own implementations, so it's a just bit more coding to implement that lacking things for my project for now ;)

**Reply by DanRibbens** [reply]
We're happy to work together on any of these requirements! I have a PR for transactions that I put on hold until we get multiple DB options further along.
There are also plans to handle document locking and handling concurrent updates—https://github.com/payloadcms/payload/discussions/2538

Our long term strategy is to make Payload 100% viable for these kinds of app framework level requirements.
